examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -99,90 +97,62 @@ public void testE2EWordCount() throws Exception {
-        e.printStackTrace();
-      }
+        // Verify outputs. Checksum is computed using SHA-1 algorithm
+        Collections.sort(outputs);
+        actualChecksum =
+                Hashing.sha1().hashString(outputs.toString(), StandardCharsets.UTF_8).toString();
+        LOG.info("Generate checksum for output data: {}", actualChecksum);
 
-      return false;
+        return actualChecksum.equals(EXPECTED_CHECKSUM);
+      } catch (IOException e) {
+        throw new RuntimeException(
+                String.format("Fail to read from path: %s", outputPath));
+      }
     }
 
-    private List<String> read(String path) {
+    private List<String> readLines(String path) throws IOException {
       List<String> readData = new ArrayList<>();
 
-      try {
         IOChannelFactory factory = IOChannelUtils.getFactory(path);
 
         // Match inputPath which may contains glob
         Collection<String> files = factory.match(path);
 
         // Read data from file paths
         for (String file : files) {
-          Reader reader = Channels.newReader(factory.open(file), StandardCharsets.UTF_8.name());
-          LineReader lineReader = new LineReader(reader);
-
-          String line;
-          while ((line = lineReader.readLine()) != null) {
-            readData.add(line);
+          try (Reader reader =
+                       Channels.newReader(factory.open(file), StandardCharsets.UTF_8.name())) {
+            readData.addAll(CharStreams.readLines(reader));
           }
-
-          reader.close();
         }
-      } catch (IOException e) {
-        e.printStackTrace();
-      }
 
       return readData;
     }
 
-    /**
-     * Generate checksum of a string collection using SHA-1 algorithm.
-     */
-    private String generateChecksum(List<String> lines) throws NoSuchAlgorithmException {
-      MessageDigest messageDigest = MessageDigest.getInstance("SHA-1");
-
-      for (String line : lines) {
-        messageDigest.update(line.getBytes(StandardCharsets.UTF_8));
-      }
-
-      return convertDecimalToHex(messageDigest.digest());
-    }
-
-    private String convertDecimalToHex(byte[] numbers) {
-      char[] digits = "0123456789ABCDEF".toCharArray();
-      StringBuilder hexNum = new StringBuilder();
-
-      for (int i = 0; i < numbers.length; i++) {
-        int num = numbers[i] & 0xFF;
-        hexNum.append(digits[num >>> 4]);
-        hexNum.append(digits[num & 0xF]);
-      }
-
-      return hexNum.toString();
+    @Override
+    public void describeTo(Description description) {
+      description
+              .appendText("Expected checksum is (")
+              .appendText(EXPECTED_CHECKSUM)
+              .appendText(")");
     }
 
     @Override
-    public void describeTo(Description description) {
-      description.appendText("Expect output checksum should be ").appendValue(EXPECTED_CHECKSUM);
+    protected void describeMismatchSafely(PipelineResult pResult, Description description) {
+      description
```

Comment:
Agree. To me, put those information in logs looks like another good solution. It makes this class stay simple and easy implement.